qp/utils.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import sys
+
+def safelog(arr, threshold=sys.float_info.epsilon):
+    """
+    Takes log of array with zeroes.
+
+    Parameters
+    ----------
+    arr: ndarray
+        Values to be logged
+    threshold: float
+        Small, positive value to replace zeros and negative numbers
+
+    Returns
+    -------
+    logged: ndarray
+        Logged values, with approximation in place of zeros and negative numbers
+    """
+    shape = np.shape(arr)
+    flat = arr.flatten()
+    logged = np.log(np.array([max(a,threshold) for a in flat])).reshape(shape)
+    return logged
+
+def calckl(qn, pn):
+    """
+    Calculates Kullback-Leibler Divergence
+
+    Parameters
+    ----------
+    qn: ndarray
```

Comment:
I changed it so the inputs are PDF objects, but it still chooses to evaluate the truth of Q and the interpolated quantile approximation of P, because a choice has to be made somewhere.  This does not permit the calculate_kl_divergence function to, say, compare two quantile approximations.  It can only compare an individual quantile approximation to a theoretical model.